src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -84,7 +84,11 @@ public function parse($expression, $names)
             return $expression;
         }
 
-        $key = $expression.'//'.implode('-', $names);
+        $keys = array_map(function($key, $value){
+            return '{'."$key:$value".'}';
+        }, array_keys($names), $names);
```

Comment:
@stof I don't want to break the original naming convention, so just extended it by including the key in the cache key.